rlkit/core/rl_algorithm.py

```diff
@@ -217,41 +217,39 @@ def pretrain(self):
 
     def collect_data(self, num_samples, resample_z_rate, update_posterior_rate, update_batch_size, add_to_enc_buffer=True):
         '''
-        collect transitions from the task
-        :param task_idx: the task ID to sample from
+        collect transitions from the current task
         :param num_samples: total number of transitions to sample
         :param resample_z_rate: how often to resample latent context z
         :param update_posterior_rate: how often to update q(z | c) from which z is sampled
-        :param update_batch_size: controls what data is used to infer the new q(z | c)
+        :param update_batch_size: how much data is used to infer the new q(z | c)
+        :param add_to_enc_buffer: whether to add collected data to encoder replay buffer
         '''
         # makes no sense to update the posterior without resampling z
         assert resample_z_rate <= update_posterior_rate
         # start from the prior
         self.reset_posterior()
-        counter = 0
+        num_collected = 0
+        update_count = 1
         for _ in range(num_samples // resample_z_rate):
-            self.collect_transitions(self.policy, num_samples=resample_z_rate, add_to_enc_buffer=add_to_enc_buffer)
-            counter += resample_z_rate
-            if counter % update_posterior_rate == 0:
+            num_collected += self.collect_transitions(self.policy, num_samples=resample_z_rate, add_to_enc_buffer=add_to_enc_buffer)
```

Comment:
collect_transitions should be renamed to collect_trajectories

Reply:
see `de1b5c3`